feat: view other user's profile pages frontend

aidandenlinger opened the pull request:
Closes #22.

User Story: As a user, I want to see information about potential roommates so I can decide to connect with them. Also, I want it to look super sleek, modern, minimal, have cool icons, knock my socks off, etc.

This PR implements the frontend for viewing other people's profile pages. This is accessible at `/viewProfile`, but doesn't currently have a direct link on the site since the "view roommates" page isn't ready yet. It is using mock user data and is not hooked up to the backend.

![Screenshot 2023-02-22 at 04-04-04 Screenshot](https://user-images.githubusercontent.com/47428697/220519372-06e16ff2-c504-4e0f-9572-7f5b36405e0d.png)

Requesting Brittany as a reviewer as she's working on the "Edit Profile" page and it makes sense to make sure our pages are relatively in sync. Feel free to pass this review to someone else if you can't review it.

Future work:
- hook up the "Request Roommate?" button to backend so it can either say "Request Roomate?", "Waiting on Response", or "Connected"
- display real property cards in the property carousel once we have mock property data
- display private contact information if the logged in user is connected to the displayed user.
- set up routing to actual users in the backend so the url can really be like `/viewProfile/<user-id>`.

bjtrieu commented on src/components/UserProfile/UserProfile.js:
```diff
@@ -0,0 +1,331 @@
+    <Grid container direction={{ xs: "column", sm: "row" }}
+      columns={2} spacing={10} margin={{ xs: 1, sm: 2 }}>
+
+      <Grid sm={2}>
+        <UserHeader user={user} />
+      </Grid>
+
+      <Grid sm={1}>
+        <UserRoommatePref user={user} />
+      </Grid>
+
+      <Grid sm={1}>
+        <UserPropertyPref user={user} />
+      </Grid>
```
- [ ] Separate the components into different files

bjtrieu commented on src/components/UserProfile/UserPropertyPref.js:
```diff
@@ -0,0 +1,68 @@
+
+				<Grid xs={2} display="flex" alignItems="center" justifyContent="center">
+					<Typography variant="h4" align="center">
+						{user.firstName}&apos;s Property Interests:
```
```suggestion
						{user.firstName}&apos;s Property Preferences:
```
david's and my page say this so just to keep it consistent

bjtrieu commented on src/components/UserProfile/PropertyCarousel.js:
```diff
@@ -0,0 +1,47 @@
+	let groupsOf3 = groupArr(interestedProp, 3);
+
+	return (
+		<Paper variant="outlined" margin={3} padding={3}>
```
```suggestion
    <Paper 
      variant="outlined" 
      sx={{
                    mt: { xs: 3, md: 6 },
                    mb: { xs: 3 },
                    p: { xs: 2, md: 3 },
                }}>
```

my paper margin and padding looks like this. This is also in the signup flow. To keep it consistent you can update it to these values. don't forget the paper components in the other files too :)

bjtrieu commented on src/components/UserProfile/UserHabits.js:
```diff
@@ -0,0 +1,82 @@
+						<TextListItem text={`Prefers roommates majoring in ${user.roommatePref.major}`} />
+						<TextListItem text={`Prefers ${user.roommatePref.schoolYear.toLowerCase()} students`} />
+						<TextListItem text={`Prefers roommates who speak ${user.roommatePref.languages.join(", ")}`} />
+						<TextListItem text={`Prefers roommates who are ${user.roommatePref.cleanliness}/5 on cleanliness`} />
```
we're not including cleanliness anymore

aidandenlinger replied:
resolved by removal of this section

bjtrieu commented on src/components/UserProfile/UserHabits.js:
```diff
@@ -0,0 +1,82 @@
+						<TextListItem text={`Prefers ${user.roommatePref.schoolYear.toLowerCase()} students`} />
+						<TextListItem text={`Prefers roommates who speak ${user.roommatePref.languages.join(", ")}`} />
+						<TextListItem text={`Prefers roommates who are ${user.roommatePref.cleanliness}/5 on cleanliness`} />
+						<TextListItem text={`Prefers roommates who are moving in around ${user.roommatePref.moveInDate}`} />
```
we're also not including move in date

aidandenlinger replied:
resolved by removal of this section

bjtrieu commented on src/components/UserProfile/UserHabits.js:
```diff
@@ -0,0 +1,82 @@
+
+				<Grid xs={2} sm={3} display="flex" justifyContent="center">
+					<List>
+						<TextListItem text={`Prefers ${user.roommatePref.gender.toLowerCase()} roommates`} />
+						<TextListItem text={`Prefers roommates who are also interested in ${user.roommatePref.hobbies.map((x) => x.toLowerCase()).join(", ")}`} />
+						<TextListItem text={`Prefers ${user.roommatePref.college} roommates`} />
+						<TextListItem text={`Prefers roommates majoring in ${user.roommatePref.major}`} />
+						<TextListItem text={`Prefers ${user.roommatePref.schoolYear.toLowerCase()} students`} />
+						<TextListItem text={`Prefers roommates who speak ${user.roommatePref.languages.join(", ")}`} />
```
This conversation should probably be moved to discord but is the user's gender, college, major, school year, and major being used as their roommate preferences? I thought this info was just for displaying on their personal information

aidandenlinger replied:
at the time, the Data Specification sheet had this information twice - once for the user and once for roommate preferences. now that it's changed to explicitly be lifestyle and habits you're right that this is invalid. I'll remove this section

bjtrieu commented on src/components/UserProfile/UserPropertyPref.js:
```diff
@@ -0,0 +1,68 @@
+
+				<Grid xs={2} display="flex" justifyContent="center">
+					<List>
+						<TextListItem text={`Wants ${Object.entries(user.propPref.amenities)
```
on Aditya's property page, there's icons for the amenities. you can include those if you want!

aidandenlinger replied:
added!

aidandenlinger commented on src/components/UserProfile/UserProfile.js:
```diff
@@ -0,0 +1,331 @@
+
+      <Grid sm={1}>
+        <UserPropertyPref user={user} />
+      </Grid>
```
resolved :)

aidandenlinger commented on src/components/UserProfile/UserProfile.js:
```diff
@@ -0,0 +1,331 @@
+
+        <Grid xs={2} md={3} display="flex" alignItems="center" justifyContent="center">
+          <Typography variant="h4" align="center">
+            {user.firstName}&apos;s Roommate Interests:
```
resolved :)

aidandenlinger commented on src/components/UserProfile/UserProfile.js:
```diff
@@ -0,0 +1,331 @@
+            <TextListItem text={`Prefers ${user.roommatePref.college} roommates`} />
+            <TextListItem text={`Prefers roommates majoring in ${user.roommatePref.major}`} />
+            <TextListItem text={`Prefers ${user.roommatePref.schoolYear.toLowerCase()} students`} />
+            <TextListItem text={`Prefers roommates who speak ${user.roommatePref.languages.join(", ")}`} />
```
at the time, the Data Specification sheet had this information twice - once for the user and once for roommate preferences. now that it's changed you're right that this is invalid. I'll remove this section

aidandenlinger commented:
implemented requested changes, most notably
- removed outdated sections
- added amenity icons
- synchronized Paper props within my component
- say "Lifestyle and Habits"

bjtrieu reviewed (approved):
looks good!

bjtrieu commented on src/components/UserProfile/PropertyCarousel.js:
```diff
@@ -0,0 +1,49 @@
+import Carousel from 'react-material-ui-carousel';
+import Stack from "@mui/material/Stack";
+
+import {paperProps} from './UserHeader.js';
```
```suggestion
import {paperProps} from './UserProfile.js';
```
looks like paperProps is being defined in UserProfile.js but you imported it from UserHeader, not sure why it's not throwing an error

aidandenlinger replied:
woah what the heck? javascript is magic, great catch, thanks

aidandenlinger replied:
fixed, not actually using any props on Paper for now because the values used look really bad on the profile page
